api/v1beta1/flinkcluster_types.go

```diff
@@ -540,7 +542,7 @@ type JobStatus struct {
 	Name string `json:"name"`
 
 	// The ID of the Flink job.
-	ID string `json:"id"`
+	ID string `json:"id,omitempty"`
```

Comment:
When a job is in the pending state, there are times when the ID value has not yet been identified. If I try to update the status with the ID nil, it is rejected because the field is required. Therefore, I changed the property to make that field optional.
```
2020-12-04T10:58:10.087Z	ERROR	controllers.FlinkCluster	Failed to update job status for new job submission	{"cluster": "test/t3", "error": "FlinkCluster.flinkoperator.k8s.io \"t3\" is invalid: status.components.job.id: Required value", "error": "FlinkCluster.flinkoperator.k8s.io \"t3\" is invalid: status.components.job.id: Required value"}
github.com/go-logr/zapr.(*zapLogger).Error
	/Users/elan/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/googlecloudplatform/flink-operator/controllers.(*ClusterReconciler).updateStatusForNewJob
	/Users/elan/dev/flink-on-k8s-operator/controllers/flinkcluster_reconciler.go:963
github.com/googlecloudplatform/flink-operator/controllers.(*ClusterReconciler).reconcileJob
	/Users/elan/dev/flink-on-k8s-operator/controllers/flinkcluster_reconciler.go:454
github.com/googlecloudplatform/flink-operator/controllers.(*ClusterReconciler).reconcile
	/Users/elan/dev/flink-on-k8s-operator/controllers/flinkcluster_reconciler.go:111
github.com/googlecloudplatform/flink-operator/controllers.(*FlinkClusterHandler).reconcile
	/Users/elan/dev/flink-on-k8s-operator/controllers/flinkcluster_controller.go:217
github.com/googlecloudplatform/flink-operator/controllers.(*FlinkClusterReconciler).Reconcile
	/Users/elan/dev/flink-on-k8s-operator/controllers/flinkcluster_controller.go:80
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/Users/elan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/Users/elan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/Users/elan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/Users/elan/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/Users/elan/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/Users/elan/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.Until
	/Users/elan/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:90
```